freqtrade/data/history/history_utils.py

```diff
@@ -1,178 +1,66 @@
-        logger.warning('Missing data at end for pair %s, data ends at %s',
-                       pair, arrow.get(pairdata[-1][0] // 1000).strftime('%Y-%m-%d %H:%M:%S'))
-
-
 def load_pair_history(pair: str,
                       timeframe: str,
-                      datadir: Path,
+                      datadir: Path, *,
                       timerange: Optional[TimeRange] = None,
                       fill_up_missing: bool = True,
                       drop_incomplete: bool = True,
                       startup_candles: int = 0,
+                      data_format: str = None,
+                      data_handler: IDataHandler = None,
                       ) -> DataFrame:
     """
     Load cached ticker history for the given pair.
 
     :param pair: Pair to load data for
     :param timeframe: Ticker timeframe (e.g. "5m")
     :param datadir: Path to the data storage location.
+    :param data_format: Format of the data. Ignored if data_handler is set.
     :param timerange: Limit data to be loaded to this timerange
     :param fill_up_missing: Fill missing values with "No action"-candles
     :param drop_incomplete: Drop last candle assuming it may be incomplete.
     :param startup_candles: Additional candles to load at the start of the period
+    :param data_handler: Initialized data-handler to use.
+                         Will be initialized from data_format if not set
     :return: DataFrame with ohlcv data, or empty DataFrame
     """
-    timerange_startup = deepcopy(timerange)
-    if startup_candles > 0 and timerange_startup:
-        timerange_startup.subtract_start(timeframe_to_seconds(timeframe) * startup_candles)
-
-    pairdata = load_tickerdata_file(datadir, pair, timeframe, timerange=timerange_startup)
-
-    if pairdata:
-        if timerange_startup:
-            _validate_pairdata(pair, pairdata, timerange_startup)
-        return parse_ticker_dataframe(pairdata, timeframe, pair=pair,
-                                      fill_missing=fill_up_missing,
-                                      drop_incomplete=drop_incomplete)
-    else:
-        logger.warning(
-            f'No history data for pair: "{pair}", timeframe: {timeframe}. '
-            'Use `freqtrade download-data` to download the data'
-        )
-        return DataFrame()
+    data_handler = get_datahandler(datadir, data_format, data_handler)
+
+    return data_handler.ohlcv_load(pair=pair,
+                                   timeframe=timeframe,
+                                   timerange=timerange,
+                                   fill_missing=fill_up_missing,
+                                   drop_incomplete=drop_incomplete,
+                                   startup_candles=startup_candles,
+                                   )
 
 
 def load_data(datadir: Path,
               timeframe: str,
-              pairs: List[str],
+              pairs: List[str], *,
```

Comment:
same note about python3.8 and syntax sugar... 
what's the purpose of this to be used here?

Reply:
This is not python.3.8 only - see the other comment for details.

The purpose is to ensure that all additional / optional arguments need to be specified using keyword arguments. 

I do this intentionally as that will 

a) allow us to reorder these arguments at will
b) improve readability on the calling site
c) avoid bugs if optional arguments are not used, but an argument "at the end of the list" is used

Reply:
LGTM 👍